libqfieldsync/project_checker.py

```diff
@@ -439,3 +445,43 @@ def check_external_layers(
             )
 
         return None
+
+    def check_project_layers_sources_actions(self) -> Optional[FeedbackResult]:
+        """Check if layers from the same GeoPackage have offline and copy actions."""
+        layer_sources_by_filename: dict[str, list[LayerSource]] = defaultdict(list)
+
+        for project_layer in self.project.mapLayers().values():
+            layer_source = LayerSource(project_layer)
+            if not layer_source.is_file:
+                continue
+
+            if layer_source.action not in [SyncAction.OFFLINE, SyncAction.COPY]:
+                continue
+
+            layer_sources_by_filename[layer_source.filename].append(
+                (layer_source.name, layer_source.action)
+            )
+
+            # Check for layers with mixed actions
+            for filename, layer_sources in layer_sources_by_filename.items():
+                has_mixed_actions = False
+                first_layer_action = layer_sources[0][1]
```

Comment:
Based on the previous comment, here we need to do something like:
```suggestion
                first_layer_source = layer_sources[0]
```

However, we need to validate that the `layer_sources` is not empty, so probably having an assert would not be a bad idea.